experiments/utils.py

```diff
@@ -366,8 +375,19 @@ def _get_disabled_experiment_names(self):
         )
 
     def set_disabled_experiments(self, names):
-        Enrollment.objects.filter(experiment__name__in=names).update(disabled=True)
-        Enrollment.objects.exclude(experiment__name__in=names).update(disabled=False)
+        for name in names:
+            experiment = experiment_manager.get_experiment(name, auto_create=False)
+            if not experiment:
+                continue
+            enrollment, _ = Enrollment.objects.get_or_create(
+                experiment=experiment,
+                user=self.user,
+                defaults={'disabled': True}
+            )
+            if not enrollment.disabled:
+                enrollment.disabled = True
+                enrollment.save()
+        Enrollment.objects.filter(user=self.user).exclude(experiment__name__in=names).update(disabled=False)
```

Comment:
isn't dangerous to change this to disable experiments for on a user basis? the previous function seem odd though

Reply:
Previous function (the first line in it) had two bugs:
1. is was performing an update for all authenticated users, when it should have done so only for the current user.
2. In case of a missing enrollment for such experiment, it failed to create a new disabled enrollment. 

This is what the change addresses. When an experiment is disabled for the current request (by a conditional), we we want it to disable the current enrollment.